docs/source/app-dev/command-deduplication.rst

```diff
@@ -0,0 +1,352 @@
+
+On some ledgers, every participant outputs the completion events for all the hosted parties, not just for the submissions that went through the participant.
+In this case, command deduplication works across participants.
+At this time, only `Daml Driver for VMware Blockchain <https://www.digitalasset.com/daml-for-vmware-blockchain/>`__ supports command deduplication across participants.
+
+    
+
+How to use command deduplication
+********************************
+
+When an application wants to effectuate a ledger change exactly once, the application must resubmit a command if an earlier submission was lost.
+However, the application typically cannot distinguish a lost submission from a Daml ledger that is just slow in processing the submissions.
+So the application should just resubmit the command until it is executed against the ledger, and rely on command deduplication to reject all duplicate submissions that do make it to the ledger.
+
+Some ledger changes can be executed at most once anyway, so no command deduplication is needed for them.
+For example, if the submitted command exercises a consuming choice on a given contract ID, this command can be accepted at most once because every contract can be archived at most once.
+All duplicate submissions of such a change will be rejected with :ref:`CONTRACT_NOT_ACTIVE <error_code_CONTRACT_NOT_ACTIVE>`.
+
+In contrast, a :ref:`Create command <com.daml.ledger.api.v1.CreateCommand>` would create a fresh contract instance of the given :ref:`template <com.daml.ledger.api.v1.CreateCommand.template_id>` for each submission that reaches the ledger (unless other constraints such as the :ref:`template preconditions <daml-ref-preconditions>` or contract key uniqueness are violated).
+Similarly, an :ref:`Exercise command <com.daml.ledger.api.v1.ExerciseCommand>` on a non-consuming choice or an :ref:`Exercise-By-Key command <com.daml.ledger.api.v1.ExercisebyKeyCommand>` may be executed multiple times if submitted multiple times.
+With command deduplication, applications can ensure such intended ledger changes are executed only once within the deduplication period, even if the application resubmits, say because it considers the earlier submissions to be lost or forgot during a crash that it had already submitted the command.
+
+As a typical example, the application wants to create exactly one contract instance of template ``T`` via the corresponding :ref:`Create command <com.daml.ledger.api.v1.CreateCommand>`.
+We assume that the ledger is correctly set up so that the Create command should be accepted, i.e., Daml packages are uploaded and vetted, the parties are allocated, and the application has a valid access token for the submission.
+
+
+Known processing time bounds
+============================
+
+For now, we assume that the application knows a bound ``B`` on the processing time and forward clock drifts in the Daml ledger w.r.t. the application’s clock.
```

Comment:
spell out "with regard to"

Reply:
Also, to the degree that this is user documentation it probably not wise to introduce thought experiments and assumptions that aren't universally valid.

Reply:
Spelled out.

I'm not sure why you call this a thought experiment. In a distributed system, you cannot get exactly once semantics unless you're willing to make some assumptions, e.g., how long it takes to process something. I've rephrased it to
```
For this strategy, you must estimate a bound ``B`` on the processing time and forward clock drifts in the Daml ledger with respect to the application’s clock.
```